Tests/DatadogTests/Datadog/Mocks/LoggingFeatureMocks.swift

```diff
@@ -189,14 +191,22 @@ class LogOutputMock: LogOutput {
     struct RecordedLog: Equatable {
         var level: LogLevel
         var message: String
+        var error: DDError?
         var date: Date
         var attributes = LogAttributes(userAttributes: [:], internalAttributes: [:])
         var tags: Set<String> = []
     }
 
     var recordedLog: RecordedLog? = nil
 
-    func writeLogWith(level: LogLevel, message: String, date: Date, attributes: LogAttributes, tags: Set<String>) {
-        recordedLog = RecordedLog(level: level, message: message, date: date, attributes: attributes, tags: tags)
+    func writeLogWith(level: LogLevel, message: String, error: Error?, date: Date, attributes: LogAttributes, tags: Set<String>) {
+        recordedLog = RecordedLog(
+            level: level,
+            message: message,
+            error: error.flatMap { DDError(error: $0) },
```

Comment:
⚠️  Another reason for using `DDError` in `LogOutput` signature. Now we need to make this mock smarter (by using part of the real output logic), to make it work. Mapping to `DDError` as soon as possible (in `Logger`) would fix this.